test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -139,7 +140,33 @@
                                           :fields      (into [] (for [op ops] [:expression (name op)]))})
                       (mt/formatted-rows (repeat int))
                       first
-                      (zipmap ops)))))))))
+                      (zipmap ops)))))))
+
+    (testing "with timestamptz columns"
+      (mt/test-drivers (set/intersection (mt/normal-drivers-with-feature :temporal-extract)
+                                         (mt/supports-timestamptz-type? driver/*driver*))
+        (mt/with-report-timezone-id "Asia/Ho_Chi_Minh"
+          (is (= (if (driver/supports? driver/*driver* :set-timezone)
+                   ;; drivers support set-timezone displays the result in the report-tz
+                   ;; we expect the extracted components will be in report-tz
+                   ;; for drivers that are not, extract should returns in UTC
+                   ["2004-03-19T09:19:09+07:00" 9]
+                   ;; TIMEZONE FIXME sqlserver behaves a bit strange from the rest
+                   ;; when querying datetimeoffset, the JDBC does returns the time
+                   ;; in the inserted timezone(Asia/Ho_Chi_Minh), but when we read it we
+                   ;; converted it back to UTC.
+                   ;; So technically we could make sqlserver display datetimeoffset in `report-tz`
+                   ;; then the extract hour will make more sense
+                   ["2004-03-19T02:19:09Z" (case driver/*driver*
+                                            :sqlserver 9
+                                            2)])
+                 (->> (mt/mbql-query times {:expressions {"hour" [:get-hour $dt_tz]}
+                                             :fields      [$dt_tz [:expression "hour"]]
+                                             :filter      [:= $index 1]
+                                             :limit       1})
```

Comment:
Should we test minutes too? You can use `Asia/Kabul` for this, since it's +4:30 all year round

Reply:
Actually now that I think about it, why don't we test all units hours too? The timezone is significant for returning days, weeks, months, and years too

Reply:
I Initially did that, but think it's too overkill.
maybe it's a good idea after all?
https://github.com/metabase/metabase/pull/26824/commits/6745c441822207521df7a6ed0b144640593a72d5

Reply:
I think it's valuable as they are separate code paths, and potentially the implementations could be very different depending on the driver. You'd want to test a datetime and timezone where the difference in hours matters though, for example on the last day of the year

Reply:
I see you changed updated it by adding/subtracting. Why not just use a literal? Then you wouldn't be limited by drivers that support date-arithmetic (which just happens to overlap nicely with set-timezone and temporal-extract, but not necessarily for 3rd party drivers)

Reply:
There are problems with the literal..turns out it only works nicely with LocalDatetime, but not offsetdatetime.

I did tried it here https://github.com/metabase/metabase/pull/26824/commits/b07b9f1cc8e912129a36671d67dd59b40e37c866.

snowflake, vertical, oracle, sqlserver were failing.

I plan to fix the literal in a different PR, since I think it's lower priority than making it work correctly with a column.

Reply:
Okay, still date-arithmetic should not be required to test temporal-extract behaviour. Perhaps leave a FIXME comment until this is fixed

Reply:
One last nitpick, I still think we should test minutes. You could do this by using "Asia/Kabul" instead of "Asia/Ho_Chi_Minh"